scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,50 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def test_gttrf_gttrs():
+    np.random.seed(42)
+    n = 10
+    for index, dtype in enumerate(DTYPES):
+        atol = 10**-(np.finfo(dtype).precision-1)
+        if index < 2:
+            du = np.random.rand(n-1)
+            d = np.random.rand(n)
+            dl = np.random.rand(n-1)
+        else:
+            du = np.random.rand(n-1) * 1j
+            d = np.random.rand(n) * 1j
+            dl = np.random.rand(n-1) * 1j
+
+        diag_cpy = np.array([dl.copy(), d.copy(), du.copy()])
+
+        A = np.diag(d) + np.diag(dl, -1) + np.diag(du, 1)
+        x = np.random.rand(n)
+        b = A @ x
+
+        dgttrf = get_lapack_funcs('gttrf', dtype=dtype)
+        dgttrs = get_lapack_funcs('gttrs', dtype=dtype)
+
+        _dl, _d, _du, du2, ipiv, info = dgttrf(dl, d, du)
+
+        np.testing.assert_allclose(dl, diag_cpy[0], atol=atol)
+        np.testing.assert_allclose(d, diag_cpy[1], atol=atol)
+        np.testing.assert_allclose(du, diag_cpy[2], atol=atol)
+        
+        b_cpy = b.copy()
+        x_dgttrs, info = dgttrs(_dl, _d, _du, du2, ipiv, b)
+        np.testing.assert_allclose(x, x_dgttrs, atol=atol)
+        np.testing.assert_allclose(b, b_cpy, atol=atol)
+        
+
+        dl_malformatted = np.append(dl, [1])
+        d_malformatted = np.append(d, [1])
+        du_malformatted = np.append(du, [1])
+        with assert_raises(ValueError):
+            dgttrf(dl_malformatted, d, du)
+        with assert_raises(ValueError):
+            dgttrf(dl, d_malformatted, du)
+        with assert_raises(ValueError):
+            dgttrf(dl, d, du_malformatted)
+
```

Comment:
Please also add a test for what happens when the input matrix is singular (e.g. one row is all zeros). The routine should fail gracefully, perhaps returning nonzero `info`.

Reply:
It currently does not, but I added a test for that when d is all zeroes.

Reply:
@swallan if it doesn't fail gracefully then how does it fail?